cfme/tests/intelligence/reports/test_tenant_quota_reports.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+
+property_mapping = {
+    "cpu": "Allocated Virtual CPUs",
+    "memory": "Allocated Memory in GB",
+    "storage": "Allocated Storage in GB",
+    "template": "Allocated Number of Templates",
+    "vm": "Allocated Number of Virtual Machines",
+}
+
+
+@pytest.fixture(scope="module")
+def set_and_get_tenant_quota(appliance):
+    root_tenant = appliance.collections.tenants.get_root_tenant()
+    tenant_quota_data = {
+        "cpu_cb": True,
+        "cpu": "10",
+        "memory_cb": True,
+        "memory": "50.0",
+        "storage_cb": True,
+        "storage": "150.0",
+        "template_cb": True,
+        "template": "5",
+        "vm_cb": True,
+        "vm": "15",
+    }
+
+    root_tenant.set_quota(**tenant_quota_data)
+
+    data = dict()
+    for key, value in property_mapping.items():
+        suffix = "GB" if "GB" in value else "Count"
+        data[value] = '{data_value} {suffix}'.format(
+            data_value=tenant_quota_data[key], suffix=suffix)
+
+    yield data
+
+    reset_data = {
```

Comment:
I'm thinking the root tenant's quota should be read before modification, and reset to its state before the test. It's possible the appliances are modified by the session in order to implement a quota for the tests in general.